packages/admin-ui/src/search/hooks/useSearchState.ts

```diff
@@ -0,0 +1,73 @@
+import type { ChangeEventHandler } from 'react'
+import { useCallback } from 'react'
+import { useDebouncedCache } from '@vtex/admin-ui-hooks'
+
+const DEFAULT_TIMEOUT_MS = 250
+
+export function useSearchState(
+  params: UseSearchStateParams = {
+    initialValue: '',
+    timeoutMs: DEFAULT_TIMEOUT_MS,
+  }
+): SearchFormStateReturn {
+  const {
+    initialValue = '',
+    defaultValue = '',
+    timeoutMs = DEFAULT_TIMEOUT_MS,
+  } = params
+
+  const [value, debouncedValue, setValue] = useDebouncedCache({
+    initialState: initialValue,
+    timeoutMs,
+  })
+
+  const onClear = useCallback(() => {
+    setValue(defaultValue)
+  }, [value])
+
+  const onChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+    (event) => {
+      setValue(event.target.value)
+    },
+    []
+  )
+
+  const getInputProps = useCallback(() => {
+    return {
+      value,
+      onChange,
+      onClear,
+    }
+  }, [value, onChange])
+
+  return {
+    getInputProps,
+    debouncedValue,
+    setValue,
+    value,
+    onChange,
+    onClear,
+  }
+}
+
+export interface GetInputPropsReturn {
+  value: string
+  onChange: ChangeEventHandler<HTMLInputElement>
+  onClear: () => void
+}
+
+export interface UseSearchStateParams {
+  initialValue?: string
+  defaultValue?: string
+  timeoutMs?: number
```

Comment:
What do you think of having this prop named as `timeout`, while also having a good jsDoc stating that it expects a value as milliseconds? I haven't seen `timeoutMs` before and properties or parameters named `timeout` are quite common across standard (and non-standard such as Axios) HTTP libraries (I know, I know, nothing to do with a UI library 😄  but why not?).